lib/kennel/unmuted_alerts.rb

```diff
@@ -66,6 +66,15 @@ def filtered_monitors(api, tag)
           end
         end
 
+        # only keep monitors that are not covered by a downtime
+        monitors.each do |m|
+          next unless m[:matching_downtimes]
+          downtime_groups = m[:matching_downtimes].select { |d| d[:active] }.flat_map { |d| d[:groups] }
+          m[:state][:groups].reject! do |k, _|
+            downtime_groups.include?(k.to_s)
```

Comment:
should behave more like line 65
a group only matches a downtime if all the dometimes tags are included in the group

Reply:
when i looked at the api response, it looked like it was already expanding the groups array into the groups that would be covered

Reply:
ok, looked that up ... groups here is an array of full names (with commas)

Reply:
... if that does not work out because of caching or datadog being slow we can use `scope` and then do the same logic as line 64 with it ...